frame_semantic_transformer/FrameSemanticTransformer.py

```diff
@@ -69,7 +69,15 @@ def __init__(
         if model_name_or_path in OFFICIAL_RELEASES:
             self.model_path = f"chanind/frame-semantic-transformer-{model_name_or_path}"
             self.model_revision = MODEL_REVISION
-        self.device = torch.device("cuda" if use_gpu else "cpu")
+        # bool is a subtype of int, so isinstance(False, int) returns True
+        if isinstance(use_gpu, int) and not isinstance(use_gpu, bool):
+            assert use_gpu in range(torch.cuda.device_count())
```

Comment:
nit: could this also include an error message like `f"GPU {use_gpu} is not available"` or something similar?

Reply:
I added another commit to this PR that includes logging a message and better error handling if the caller passes something unexpected for `use_gpu`.